advisories/0011.md

```diff
@@ -0,0 +1,57 @@
+### Mirantis Kubernetes Engine (MKE) and CVE-2024-21626: Understanding the Impact and Mitigation
+#### Release Date
+2024-02-08
+#### Overview
+This advisory addresses the impact of CVE-2024-21626, a critical vulnerability identified within runc, and its implications for the Mirantis Kubernetes Engine (MKE). Given that MKE relies heavily on the Mirantis Container Runtime (MCR), this vulnerability directly impacts its operational integrity. We aim to clarify the situation, explain the mitigation steps, and provide actionable guidance for MKE users.
+
+### Vulnerability Information
+#### CVE Identifier
+CVE-2024-21626
+#### CVSSv3.1
+8.6 (High) CVSS:3.1/AV:L/AC:L/PR:N/UI:R/S:C/C:H/I:H/A:H
+#### CWEs
+CWE-403, CWE-668
+
+#### Affected Product
+* MKE deployment requires installation of MCR.  Older versions of MCR (<= 23.0.9) contain a vulnerability in the underlying runc component. Please see [the MCR advisory](https://github.com/Mirantis/security/blob/main/advisories/0010.md) for details. Any deployment of MKE with MCR version <= 23.0.9 must upgrade to MCR 23.0.9-1 or later to mitigate this risk.
+
+* No MKE Image Update Needed for CVE-2024-21626
+  
+   While CVE-2024-21626 might appear in vulnerability scans of your MKE images due to a dependent library, an update to MKE images is not currently required. MKE's architecture mitigates the vulnerability, and there's no known exploit path within MKE. We understand this discrepancy can be confusing. To address this and enhance transparency, we're actively working to remove unnecessary references to the vulnerability from future MKE image scans. This update is planned for upcoming MKE releases.
+
+#### How to check if MKE is vulnerable
+1. [Download and config the client bundle](https://docs.mirantis.com/mke/3.7/ops/access-cluster/download-configure-client-bundle.html)
+2. Check the MCR version by
+   ```
+   docker version --format "{{range .Server.Components}}{{if eq .Name \"Engine\"}}{{.Version}}{{end}}{{end}}"
+   ```
+3. The MKE cluster is vunerable if and only if the version <= 23.0.9.
```

Comment:
Check the runc version, not Engine.

Reply:
While I understand the root cause lies with runc, MKE customers interact directly with MCR, not runc. Therefore, focusing on MCR version checks seems more relevant.

Reply:
Users will remain vulnerable if they upgrade their docker-ee package without upgrading their containerd.io package, as is possible (at least) on RPM distros. The _only_ way to tell for sure is to check the version of runc, **which docker version already conveniently outputs**

Reply:
I may add a check to `runc` and perhaps an after-upgrade check as well.  However please check https://github.com/Mirantis/security/pull/4/files#r1480921670 for my understanding of MCR as a product.

Reply:
We want users to always treat MCR as a single integrated product but we distribute it as four separate DEB/RPM packages. There are all sorts of ways that users may get their deployments into an unsupported state, intentionally or by accident. Users do weird things, make mistakes, or they might even do everything right but fall victim to a software bug during upgrade. Users are going to use these instructions to to verify that they have applied the update correctly. Instructions which incorrectly tell users they are not vulnerable if they applied the update incorrectly are worse than useless, _they put our customers at risk_.